doc/source/rllib/connector-v2.rst

```diff
@@ -0,0 +1,596 @@
+* How to manipulate the action logits coming from the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`.
+
+
+Adding past rewards to the Model's input
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to add the most recent reward to be part of the next observation going into your
+:py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`, you can write a custom
+env-to-module :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` piece like so:
+
+
+.. testcode::
+
+    import gymnasium as gym
+    import numpy as np
+    from ray.rllib.connectors.connector_v2 import ConnectorV2
+
+
+    class AddLastRewardToObservations(ConnectorV2):
+
+        # Define how the observation space will change because of this connector piece.
+        def recompute_output_observation_space(self, in_obs_space, in_act_space):
+            # For simplicity, assert input obs space is a 1D Box
+            assert isinstance(in_obs_space, gym.spaces.Box) and len(in_obs_space.shape) == 1
+            return gym.spaces.Box(
+                low=list(in_obs_space.low) + [float("-inf")],
+                high=list(in_obs_space.high) + [float("inf")],
+                shape=(in_obs_space.shape[0] + 1,),
+                dtype=in_obs_space.dtype,
+            )
+
+        # Define the actual transformation.
+        def __call__(self, *, rl_module, batch, episodes, **kwargs):
+            # Loop through all `episodes`.
+            for single_agent_episode in self.single_agent_episode_iterator(episodes):
+                # Get last reward (or 0.0 if right after `env.reset`).
+                reward = single_agent_episode.get_reward(-1, fill=0.0)
+                # Get last observation.
+                obs = single_agent_episode.get_observation(-1)
+                # Append reward to obs array.
+                new_obs = np.append(obs, reward)
+
+                # Write new observation back into the episode.
+                single_agent_episode.set_observation(new_value=new_obs, at_index=-1)
+
+            return batch
+
+
+.. tip::
+    There is already an off-the-shelf ConnectorV2 piece available for users, which performs the task of
+    adding the `N` most recent rewards and/or `M` most recent actions to the observations:
+
+    .. code-block:: python
+
+        from ray.rllib.connectors.env_to_module.prev_actions_prev_rewards import PrevActionsPrevRewards
+
+        config.env_runners(
+            env_to_module_connector=lambda env: PrevActionsPrevRewards(n_prev_rewards=N, n_prev_actions=M),
+        )
+
+
+If you plug in this custom :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` class into your algorithm config
+(`config.env_runners(env_to_module_connector=lambda env: AddLastRewardToObservations())`),
+your model should receive observations that have the most recent reward "attached" to it.
+
+Notice that in the example here, the transformed observations were written right back into the given episodes
```

Comment:
```suggestion
In the provided example, the transformed observations were written right back into the given episodes
```

Not sure the actor here in your passive voice "were written", but rewrite that section so that "X writes transformed observations into the given episodes instead of placing them into the batch."